File: AULA_07/JAVA/Matriz.java
/* Matriz.java
 * Leitura de uma matriz e
 * soma de duas matrizes, 
 * transposta delas
 * produto externo de dois vetores
 * Entrada: int [][] A, B, int []  x, y
 * Saída: double [][] C 
 * Autor: Fabrício Olivetti de França
 */

import java.io.File;
import java.io.PrintWriter;
import java.util.Scanner;
import java.util.Random;
import java.io.FileNotFoundException;

class Matriz{
	public static void main(String[] args) throws FileNotFoundException {

		Scanner leitor = new Scanner(new File("matriz1.csv")); // agora vou ler do arquivo
		leitor.useDelimiter("\\s*;\\s*|\\r?\\n");
		int n1 = leitor.nextInt();
		int m1 = leitor.nextInt();
		int [][] A = new int [n1][m1];

		for( int i=0; i<n1; i++ ){
			for( int j=0; j<m1; j++ ){
				A[i][j] = leitor.nextInt();
			}
		}
		leitor.close();

		leitor = new Scanner(new File("matriz2.csv")); // agora vou ler do arquivo
		leitor.useDelimiter("\\s*;\\s*|\\r?\\n");
		int n2 = leitor.nextInt();
		int m2 = leitor.nextInt();
		int [][] B = new int [n2][m2];

		for( int i=0; i<n2; i++ ){
			for( int j=0; j<m2; j++ ){
				B[i][j] = leitor.nextInt();
			}
		}
		leitor.close();

		leitor = new Scanner(new File("vetor1.csv")); // agora vou ler do arquivo
		leitor.useDelimiter("\\s*;\\s*|\\r?\\n");
		int n3 = leitor.nextInt();
		int [] x= new int [n3];

		for( int i=0; i<n3; i++ ){
			x[i] = leitor.nextInt();     
		}
		leitor.close();

		leitor = new Scanner(new File("vetor2.csv")); // agora vou ler do arquivo
		leitor.useDelimiter("\\s*;\\s*|\\r?\\n");
		int n4 = leitor.nextInt();
		int [] y= new int [n4];

		for( int i=0; i<n4; i++ ){
			y[i] = leitor.nextInt();     
		}
		leitor.close();


		// C = A + B
		int [][] C = new int [n1][m1];
		if( n1 == n2 && m1 == m2 ){ // só faço a soma se as dimensões forem iguais
			for( int i=0;i<n1;i++ ){
				for( int j=0;j<m1;j++ ){
					C[i][j] = A[i][j] + B[i][j];
				}
			}
		}
		PrintWriter saida = new PrintWriter("somaMatriz.csv"); // vamos gravar em um arquivo!
		saida.println(n1+";"+m1);
		for( int i=0; i<n1;i++ ){
			for( int j=0; j<m1-1; j++ ){
				saida.print(C[i][j]+";");
			}
			saida.println(C[i][m1-1]);
		}
		saida.close();

		// Transposta de A
		int [][] T = new int [m1][n1]; // dimensões invertidas de A
		for( int i=0;i<n1;i++ ){
			for( int j=0;j<m1;j++ ){
				T[j][i] = A[i][j];
			}
		}

		saida = new PrintWriter("transposta.csv");
		saida.println(m1+";"+n1);
		for( int i=0; i<m1;i++ ){
			for( int j=0; j<n1-1; j++ ){
				saida.print(T[i][j]+";");
			}
			saida.println(T[i][n1-1]);
		}
		saida.close();

		// produto externo x e y
		int [][] D = new int [n3][n4]; // D vai ter a dimensão de x e y
		for( int i=0;i<n3;i++ ){
			for( int j=0;j<n4;j++ ){
				D[i][j] = x[i]*y[j];
			}
		}
		saida = new PrintWriter("produtoexterno.csv");
		saida.println(n3+";"+n4);
		for( int i=0; i<n3;i++ ){
			for( int j=0; j<n4-1; j++ ){
				saida.print(D[i][j]+";");
			}
			saida.println(D[i][n4-1]);
		}
		saida.close();

	}
}

File: AULA_07/Python/Matriz.py
# -*- coding: cp1252 -*-
'''
ProdMatriz.py
 * Multiplica duas matrizes e imprime o resultado
 * 
 * Entrada: int [][] A, B
 * Saída: double [][] C 
 * Autor: Fabrício Olivetti de França
 *
 * DICA:
 * verifique se as dimensões estão corretas!
'''

f = open("matriz1.csv")
n1,m1 = map(int,f.readline().split(";"))
A = []
for i in range(n1):
    A.append(map(int,f.readline().split(";")))
f.close()

f = open("matriz2.csv")
n2,m2 = map(int,f.readline().split(";"))
B = []
for i in range(n2):
    B.append(map(int,f.readline().split(";")))
f.close()

f = open("vetor1.csv")
n3 = int(f.readline())
x = map(int,f.readline().split(";"))
f.close()

f = open("vetor2.csv")
n4 = int(f.readline())
y = map(int,f.readline().split(";"))
f.close()

# C = A+B
C = [[0]*m1 for _ in range(n1)]
if n1==n2 and m1==m2:
    for i in range(n1):
        for j in range(m1):
            C[i][j] = A[i][j]+B[i][j]
    fw = open("somaMatriz.csv","w")
    print >>fw, C
    fw.close()

# Transposta de A
T = [[0]*n1 for _ in range(m1)]
for i in range(n1):
    for j in range(m1):
        T[j][i] = A[i][j]
fw = open("transposta.csv","w")
print >>fw, T
fw.close()

# produto externo x e y
D = [[0]*n4 for _ in range(n3)]
for i in range(n3):
    for j in range(n4):
        D[i][j] = x[i]*y[j]
fw = open("produtoexterno.csv","w")
print >>fw, D
fw.close()
